packages/core/src/rules/common/assertions/utils.ts

```diff
@@ -105,10 +105,10 @@ export function buildSubjectVisitor(
         for (const property of properties) {
           // we can have resolvable scalar so need to resolve value here.
           const value = isRef(node[property]) ? resolve(node[property])?.node : node[property];
-          runAssertion(value, assert, location.child(property), report);
+          runAssertion(value, node[property], assert, location.child(property), report);
         }
       } else {
-        runAssertion(Object.keys(node), assert, location.key(), report);
+        runAssertion(Object.keys(node), Object.keys(node), assert, location.key(), report);
```

Comment:
should it use Object.values?

Reply:
I believe it shouldn't, because we use run assertions on keys when property is not specified, no assertions expects an array of values